src/eredis_client.erl

```diff
@@ -463,6 +469,9 @@ do_sync_command(Socket, Command) ->
                 {ok, <<"+OK\r\n">>} ->
                     set_transport_opts(Socket, [{active, once}]),
                     ok;
+                %% {ok, <<"-WRONGPASS invalid username-password pair or user is disabled.\r\n">>}
+                {ok, ErrorMsg} ->
+                    {error, ErrorMsg};
                 Other ->
```

Comment:
Now this clause only matches posix errors, maybe just match on `{error, Reason}` and return `{error, Reason}`.
NOTE: this is a change of the API contract, we need to search all callers to see if this change of return value is OK for them.

Reply:
Good news, so far. No one parses `Reason` field.